crates/holochain_integrity_types/src/rate_limit.rs

```diff
@@ -0,0 +1,101 @@
+//! Rate limiting data types
+
+use holochain_serialized_bytes::prelude::*;
+
+use crate::{Create, CreateLink, Delete, Entry, Update};
+
+/// Input to the `weigh` callback. Includes an "unweighed" header, and Entry
+/// if applicable.
+#[derive(Clone, PartialEq, Serialize, Deserialize, SerializedBytes, Debug)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+pub enum WeighInput {
+    /// A Link to be weighed
+    Link(CreateLink<()>),
+    /// A new Entry to be weighed
+    Create(Create<()>, Entry),
+    /// An updated Entry to be weighed
+    Update(Update<()>, Entry),
+    /// An Entry deletion to be weighed
+    Delete(Delete<()>),
+}
+
+/// A bucket ID, for rate limiting
+pub type RateBucketId = u8;
+
+/// The weight of this header, for rate limiting
+pub type RateUnits = u8;
+
+/// The normalized total size of this header, for rate limiting
+pub type RateBytes = u8;
+
+/// The amount that a bucket is "filled"
+pub type RateBucketCapacity = u32;
```

Comment:
This type isn't used anywhere, is it needed for the future?

Reply:
Yes, it's used in the actual implementation, in subsequent PRs